src/action_management/src/actionmanagement.c

```diff
@@ -65,38 +113,67 @@ int action_item(game_t *g, action_t *a, item_t *i)
 	assert(g->current_player); // assumes game_t has a field for current player
 	assert(a);
 	assert(i);
-	if (a->kind != ITEM_NPC) {
-		fprintf(stderr, "The action provided is not of the correct type");
+	if (a->kind != ITEM) {
+		fprintf(stderr, "The action provided is not of the correct type.\n");
 		return FAILURE;
 	}
 	/* function needs to be implemented by game state
 	 * takes in a player, npc, and item
 	 * removes object from inventory
 	 * frees object if necessary
-	 * returns 1 if success, 0 if failed
+	 * returns 0 if success, 1 if failed
 	 */
 	int given = remove_inventory_object(g->current_player, i);
+    // THIS IS A non-existent FUNCTION we depend on GAME STATE FOR
 	//ex if "OPEN DOOR", game state needs to change the state of the door
 	if (given != SUCCESS) {
-		fprintf(stderr, "Object could not be moved from inventory");
+		fprintf(stderr, "Object could not be moved from inventory.\n");
 		return FAILURE;
 	}
 	return SUCCESS;
 
 }
 
+//KIND 2
+/* See actionmanagement.h */
+int action_direction(game_t *g, action_t *a, direction_t *d)
+{
+    assert(g);
+    assert(a);
+    assert(n);
+    if (a->kind != DIRECTION) {
+        fprintf(stderr, "The action provided is not of the correct type.\n");
+        return FAILURE;
+    }
+
+    int moved = player_move(d); // THIS IS A non-existent FUNCTION we depend on GAME STATE FOR
+    if (moved == SUCCESS) {
+        return SUCCESS;
+    } else {
+        fprintf(stderr, "Player was unable to move.\n");
+        return FAILURE;
+    }
+}
+
 //KIND 3
 /* See actionmanagement.h */
-int action_npc(action_t *a, npc_t *n)
+int action_npc(game_t *g, action_t *a, npc_t *n)
```

Comment:
Again, I don't think game_t is necessary b/c it isn't ever used within the function
```suggestion
int action_npc(action_t *a, npc_t *n)
```

Reply:
I agree with this, but maybe we should still have a player parameter.

Reply:
we don't yet fully know what will have to be altered when you talk to an npc, but I say it's safe and also consistent with the other functions just to leave the game state in.

also, if you only have the npc as an input you won't be able to check for validity to see if that npc is actually in the room with you -- the game struct holds all the rooms you are in